### PR TITLE
feat: Add logging util and CLI params for logging output

### PR DESCRIPTION
**Pull request by mkmeral**
## Description
This PR adds logging system to the Strands CLI. The implementation includes:

1. Command-line arguments for configuring logging:
   - `--log-level`: Set logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)
   - `--log-file`: Specify output file path for logs

2. A new utility module `logging_utils.py` with functions for:
   - Configuring logging with file output
   - Getting available log levels
   - Checking logging status
   - Setting log levels for specific modules

3. Integration into the main CLI entry point to initialize logging based on user preferences

4. Comprehensive unit tests in `tests/utils/test_logging_utils.py` covering:
   - Log level validation
   - File handler creation
   - Directory creation
   - Exception handling
   - Logger state management
   - All edge cases and error conditions

This enhancement will help users debug issues, track agent behavior, and maintain operation logs for production deployments.

## Related Issues
[No specific issue reference provided - please add if applicable]

## Documentation PR
[Link to associated documentation PR when created]

## Type of Change
- New feature

## Testing
The changes have been thoroughly tested with:

* `hatch fmt --linter` - All linting checks pass
* `hatch fmt --formatter` - Code formatting is consistent
* `hatch test --all` - Full test suite passes

Additionally, manual testing with various logging configurations confirms the functionality:
```bash
# Test with default logging (INFO level)
strands --log-file=strands.log

# Test with debug level
strands --log-level=DEBUG --log-file=debug.log

# Verify log file is created with appropriate content
cat debug.log
```

See example
```
2025-05-25 11:06:53,210 - root - INFO - Logging configured with level: DEBUG
2025-05-25 11:06:53,210 - root - INFO - Log file: /Users/murmeral/.strands/logs/2025-05-25_11-06-52-DEBUG.log
2025-05-25 11:06:53,210 - root - INFO - Strands CLI started with log level DEBUG
2025-05-25 11:06:53,210 - root - INFO - Log file: /Users/murmeral/.strands/logs/2025-05-25_11-06-52-DEBUG.log
2025-05-25 11:06:53,210 - strands.models.bedrock - DEBUG - config=<{'model_id': 'us.anthropic.claude-3-7-sonnet-20250219-v1:0', 'max_tokens': 64000, 'additional_request_fields': {'thinking': {'type': 'enabled', 'budget_tokens': 2048}}}> | initializing
2025-05-25 11:06:53,211 - botocore.hooks - DEBUG - Changing event name from creating-client-class.iot-data to creating-client-class.iot-data-plane
...
```

## Checklist
- [x] I have read the CONTRIBUTING document
- [x] I have added tests that prove my fix is effective or my feature works
- [ ] I have updated the documentation accordingly
- [ ] I have added an appropriate example to the documentation to outline the feature
- [x] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice

### CODE REVIEW

File: src/strands_agents_builder/utils/logging_utils.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
+"""
+Utility functions for configuring and managing logging in the Strands CLI.
```
Comment (awsarron) [inline src/strands_agents_builder/utils/logging_utils.py:3]: nit: Strands Agent Builder*

File: src/strands_agents_builder/utils/logging_utils.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env python3
```
Comment (awsarron) [inline src/strands_agents_builder/utils/logging_utils.py:1]: think the shebang can be removed

File: src/strands_agents_builder/strands.py [hunk 3]
```diff
@@ -69,8 +71,26 @@ def main():
...
+        "--log-level",
+        type=str,
+        default="INFO",
+        choices=get_available_log_levels(),
+        help="Set the logging level (DEBUG, INFO, WARNING, ERROR, CRITICAL)",
+    )
+    parser.add_argument(
+        "--log-file",
+        type=str,
+        help="Path to log file. If not specified, logging is disabled",
```
Comment (awsarron) [inline src/strands_agents_builder/strands.py:84]: could it perhaps be useful to output on stderr in some cases if log file is not defined?

perhaps default log-level to INFO only if log-file is passed and log-level is not passed? Otherwise if log-level and log-file are not passed, don't enable logging. If only log-level is passed, perhaps output logs to stderr as `logging` does by default.

File: src/strands_agents_builder/strands.py [hunk 4]
```diff
@@ -69,8 +71,26 @@ def main():
...
+    )
     args = parser.parse_args()
 
+    # Configure logging
+    configure_logging(log_level=args.log_level, log_file=args.log_file)
+    if args.log_file:
+        logging.info(f"Strands CLI started with log level {args.log_level}")
```
Comment (awsarron) [inline src/strands_agents_builder/strands.py:91]: should we be using the root logger `logging` here, or instead get a logger for `strands_agents_builder` ?

File: src/strands_agents_builder/utils/logging_utils.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+    # Configure root logger
+    logging.basicConfig(
+        level=numeric_level,
+        format=log_format,
+        handlers=handlers,
+        force=True,  # Force reconfiguration
+    )
+
+    # Configure specific Strands loggers
+    loggers = ["strands", "strands.agent", "strands.models", "strands.tools", "strands_agents_builder"]
```
Comment (awsarron) [inline src/strands_agents_builder/utils/logging_utils.py:67]: `strands.agent`, `strands.models`, `strands.tools` might be redundant if `strands` is configured.

we may also want to configure `strands_tools`.

File: src/strands_agents_builder/utils/logging_utils.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+
+
+def get_available_log_levels() -> List[str]:
+    """
+    Returns a list of available logging levels.
+
+    Returns:
+        List of log level names
+    """
+    return ["DEBUG", "INFO", "WARNING", "ERROR", "CRITICAL"]
```
Comment (awsarron) [inline src/strands_agents_builder/utils/logging_utils.py:85]: do we need this function? Seems redundant when `logging` is what actually defines the log levels

File: src/strands_agents_builder/utils/logging_utils.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+
+
+def get_logging_status() -> Dict[str, Any]:
```
Comment (awsarron) [inline src/strands_agents_builder/utils/logging_utils.py:88]: this function isn't used

File: src/strands_agents_builder/utils/logging_utils.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+    # Get status of strands specific loggers
+    for logger_name in ["strands", "strands.agent", "strands.models", "strands.tools", "strands_agents_builder"]:
+        logger = logging.getLogger(logger_name)
+        status["strands_loggers"][logger_name] = logging.getLevelName(logger.level)
+
+    return status
+
+
+def set_log_level_for_module(module_name: str, log_level: Union[str, int]) -> None:
```
Comment (awsarron) [inline src/strands_agents_builder/utils/logging_utils.py:115]: this function isn't used

**Review by awsarron — CHANGES_REQUESTED** [review]

File: src/strands_agents_builder/utils/logging_utils.py (hunk 6 above)
Comment (mkmeral) [reply src/strands_agents_builder/utils/logging_utils.py:86]: This is for the CLI interface
